feat: document latch develop (#231)
latchbio/latch: docs/source/basics/local_development.md

# Development and debugging

> When developing a workflow, its useful to run your task functions before running the entire workflow in the cloud so that you can debug environment and logical issues. However, since tasks run in a different environment than your local computer (namely, the one defined in your Dockerfile(s)), it isn't always feasible or instructive to run your workflows locally.

Reviewer: ```suggestion
When developing a workflow, it's useful to run each task function before running the entire workflow in the cloud to debug environment and logical issues. However, since tasks run in the environment defined by the Dockerfile rather than on your computer, it isn't trivial to run them locally.
```
the phrasing still sucks but can't rewrite it any better without a complete redo which would change voice and obviously fall out of line with the rest of the document

> To address this, the Latch SDK comes with a command that allows you to run tasks and debug your environment without having to go through the web or constantly reregister your workflow. Navigate to a workflow directory you would like to work on and run

Reviewer: ```suggestion
To address this, the Latch SDK comes with a command that allows you to run tasks and debug your environment without having to constantly reregister your workflow and run it through the UI.

<!-- new section here? -->

Navigate to a workflow directory you would like to work on and run
```

$ latch develop .
```

This command will drop you into a shell in which you can run your code and inspect your environment.

Output:

```console
$ latch develop .

> ...

Reviewer: replace with actual output

)
```

You can execute the script in `latch develop` like so:

```console
>>> python3 scripts/test_task.py
```

You can execute any python code in your workflow environment. Functions and scripts do not have to be latch specific so you can test library code or binaries as well. Think of the environment as a snapshot of the computer your tasks run on.

## Notes on the test environment

> It is important that any changes to the code are done on your local machine -- these changes will be synced into the latch develop environment and saved on your local computer as well. Changes made directly in the latch develop environment are not saved and are not synced back to your local computer. Moreover, they may be overwritten by the sync process.

Reviewer: ```suggestion
It is important that any changes to the code are done on your local machine --- these changes will be synced into the latch develop environment and saved on your local computer as well. Changes made directly in the latch develop environment are not saved and are not synced back to your local computer. Moreover, they may be overwritten by the sync process.
```

> We use `rsync` to bring changes from your local workflow directory to the latch develop environment. We recursively copy changes to the `/root` directory in the development environment. For example, this line in the default docker image creates the `wf` directory in the cloud environment:

Reviewer: mentioning dockerfile modification here makes it seem like `rsync` is not automatic

```Dockerfile
...

> COPY wf /root/wf

Reviewer: ```suggestion
copy wf /root/wf
```
vanity
